### PR TITLE
Fix tests for support msa branch

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTargetTest.java [hunk 1]
```diff
@@ -284,4 +292,57 @@ public void createChildrenAfterDisabling() {
...
+        AuthenticationProvider.TokenProvider tokenProvider = mock(AuthenticationProvider.TokenProvider.class);
+        AuthenticationProvider authenticationProvider = new AuthenticationProvider(AuthenticationProvider.Type.MSA, "key1", tokenProvider);
+        AnalyticsTransmissionTarget.addAuthenticationProvider(authenticationProvider);
+        assertEquals(authenticationProvider, AnalyticsTransmissionTarget.sAuthenticationProvider);
+
+        /* No actions are prepared with no CommonSchemaLog. */
+        AnalyticsTransmissionTarget.getChannelListener().onPreparingLog(mock(Log.class), "test");
+
+        /* Call prepare log. */
+        final ProtocolExtension protocol = mock(ProtocolExtension.class);
```
Comment (guperrot) [inline sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTargetTest.java]: No need to mock these, you can check other unit tests.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/OneCollectorIngestion.java [hunk 2]
```diff
@@ -137,15 +137,17 @@ public ServiceCall sendAsync(String appSecret, UUID installId, LogContainer logC
...
-            List<String> ticketKeys = ((CommonSchemaLog) log).getExt().getProtocol().getTicketKeys();
-            if (ticketKeys != null) {
-                for (String ticketKey : ticketKeys) {
-                    String token = TicketCache.getInstance().getTicket(ticketKey);
-                    if (token != null) {
-                        try {
-                            tickets.put(ticketKey, "d:" + token);
-                        } catch (JSONException e) {
-                            throw new RuntimeException(e);
+            if (log instanceof CommonSchemaLog) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/OneCollectorIngestion.java]: This if check will never be false in reality.

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/ingestion/OneCollectorIngestionTest.java [hunk 3]
```diff
@@ -118,6 +154,58 @@ public ServiceCall answer(InvocationOnMock invocation) {
         verify(httpClient).reopen();
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    public void passTickets() throws Exception {
+
+        /* Build some payload. */
+        final CommonSchemaLog log1 = mock(CommonSchemaLog.class);
```
Comment (guperrot) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/ingestion/OneCollectorIngestionTest.java]: I think we have MockCommonSchemaLog.

Comment (MatkovIvan) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/ingestion/OneCollectorIngestionTest.java]: Only in instrumented tests, not unit

**Pull request merged**